### ISSUE TITLE
read of 1 byte, stack-use-after-scope in GlobalTrajectoryParameters::charge()

### ISSUE THREAD

**Issue by davidlt** [cms-sw/cmssw#21105]
CMSSW: `CMSSW_10_0_ASAN_X_2017-10-30-2300`
SCRAM_ARCH: `slc6_amd64_gcc700`
Workflow: `3.0`
Step: 3

Detailed report:
```
31-Oct-2017 18:58:22 CET  Successfully opened file file:step2.root
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 at 31-Oct-2017 19:00:34.551 CET
=================================================================
==24864==ERROR: AddressSanitizer: stack-use-after-scope on address 0x7fff5a3889e0 at pc 0x7f6ab43ffd51 bp 0x7fff5a385a60 sp 0x7fff5a385a58
READ of size 1 at 0x7fff5a3889e0 thread T0
    #0 0x7f6ab43ffd50 in GlobalTrajectoryParameters::charge() const /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h:90
    #1 0x7f6ab4411023 in FreeTrajectoryState::charge() const /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/TrackingTools/TrajectoryState/interface/FreeTrajectoryState.h:91
    #2 0x7f6ab440d7d4 in ClosestApproachInRPhi::calculate(FreeTrajectoryState const&, FreeTrajectoryState const&) /mnt/build/davidlt/debug/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/TrackingTools/PatternTools/src/ClosestApproachInRPhi.cc:29
    #3 0x7f6a7f6b89da in V0Fitter::fitAll(edm::Event const&, edm::EventSetup const&, std::vector<reco::VertexCompositeCandidate, std::allocator<reco::VertexCompositeCandidate> >&, std::vector<reco::VertexCompositeCandidate, std::allocator<reco::VertexCompositeCandidate> >&) /mnt/build/davidlt/debug/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/RecoVertex/V0Producer/src/V0Fitter.cc:161
    #4 0x7f6a7f6980a7 in V0Producer::produce(edm::Event&, edm::EventSetup const&) /mnt/build/davidlt/debug/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/RecoVertex/V0Producer/src/V0Producer.cc:78
    #5 0x7f6ad10842e3 in edm::stream::EDProducerAdaptorBase::doEvent(edm::EventPrincipal const&, edm::EventSetup const&, edm::ActivityRegistry*, edm::ModuleCallingContext const*) (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x6b32e3)
    #6 0x7f6ad0cdd33c in edm::WorkerT<edm::stream::EDProducerAdaptorBase>::implDo(edm::EventPrincipal const&, edm::EventSetup const&, edm::ModuleCallingContext const*) (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x30c33c)
    #7 0x7f6ad0b6f09d in decltype ({parm#1}()) edm::convertException::wrap<bool edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(bool edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x19e09d)
    #8 0x7f6ad0b6f522 in bool edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x19e522)
    #9 0x7f6ad0b7703c in std::__exception_ptr::exception_ptr edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::MyPrincipal const&, edm::EventSetup const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x1a603c)
    #10 0x7f6ad0b79f29 in edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute() (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x1a8f29)
    #11 0x7f6acef0c055 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all(tbb::task&, tbb::task*) ../../src/tbb/custom_scheduler.h:509
    #12 0x7f6ad0e29f5f in edm::EventProcessor::readAndProcessEvents() (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x458f5f)
    #13 0x7f6ad0e58346 in edm::EventProcessor::runToCompletion() (/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/lib/slc6_amd64_gcc700/libFWCoreFramework.so+0x487346)
    #14 0x412f67 in main::{lambda()#1}::operator()() const (/cvmfs/cms-ib.cern.ch/nweek-02496/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/bin/slc6_amd64_gcc700/cmsRun+0x412f67)
    #15 0x40d322 in main (/cvmfs/cms-ib.cern.ch/nweek-02496/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/bin/slc6_amd64_gcc700/cmsRun+0x40d322)
    #16 0x7f6acd8b9d5c in __libc_start_main (/lib64/libc.so.6+0x1ed5c)
    #17 0x40da88  (/cvmfs/cms-ib.cern.ch/nweek-02496/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/bin/slc6_amd64_gcc700/cmsRun+0x40da88)

Address 0x7fff5a3889e0 is located in stack of thread T0 at offset 11120 in frame
    #0 0x7f6a7f6b6ebf in V0Fitter::fitAll(edm::Event const&, edm::EventSetup const&, std::vector<reco::VertexCompositeCandidate, std::allocator<reco::VertexCompositeCandidate> >&, std::vector<reco::VertexCompositeCandidate, std::allocator<reco::VertexCompositeCandidate> >&) /mnt/build/davidlt/debug/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/RecoVertex/V0Producer/src/V0Fitter.cc:89

  This frame has 130 object(s):
    [32, 33) 'addp4'
    [96, 97) '<unknown>'
    [160, 161) '<unknown>'
    [224, 225) '<unknown>'
    [288, 289) '<unknown>'
    [352, 353) '<unknown>'
    [416, 417) '<unknown>'
    [480, 484) '<unknown>'
    [544, 548) '<unknown>'
    [608, 612) '<unknown>'
    [672, 680) 'iTk'
    [736, 744) '<unknown>'
    [800, 808) 'tmpTransient'
    [864, 872) '<unknown>'
    [928, 936) '<unknown>'
    [992, 1000) '<unknown>'
    [1056, 1064) '<unknown>'
    [1120, 1128) '<unknown>'
    [1184, 1192) '<unknown>'
    [1248, 1256) '<unknown>'
    [1312, 1320) '<unknown>'
    [1376, 1384) 'trajPlus'
    [1440, 1448) 'trajMins'
    [1504, 1512) 'iTrack'
    [1568, 1576) '<unknown>'
    [1632, 1640) 'piPlusE'
    [1696, 1704) 'piMinusE'
    [1760, 1768) 'protonE'
    [1824, 1832) 'antiProtonE'
    [1888, 1896) 'kShortETot'
    [1952, 1960) 'lambdaEtot'
    [2016, 2024) 'lambdaBarEtot'
    [2080, 2088) '<unknown>'
    [2144, 2152) '<unknown>'
    [2208, 2216) '<unknown>'
    [2272, 2280) '<unknown>'
    [2336, 2344) '<unknown>'
    [2400, 2408) '<unknown>'
    [2464, 2472) '<unknown>'
    [2528, 2536) '<unknown>'
    [2592, 2600) '<unknown>'
    [2656, 2664) '<unknown>'
    [2720, 2728) '<unknown>'
    [2784, 2792) '<unknown>'
    [2848, 2856) '<unknown>'
    [2912, 2920) '<unknown>'
    [2976, 2984) '<unknown>'
    [3040, 3048) '<unknown>'
    [3104, 3112) '<unknown>'
    [3168, 3176) '<unknown>'
    [3232, 3240) '<unknown>'
    [3296, 3304) '<unknown>'
    [3360, 3368) '<unknown>'
    [3424, 3432) '<unknown>'
    [3488, 3496) '<unknown>'
    [3552, 3560) '<unknown>'
    [3616, 3624) '<unknown>'
    [3680, 3688) '<unknown>'
    [3744, 3752) '<unknown>'
    [3808, 3816) '<unknown>'
    [3872, 3880) '<unknown>'
    [3936, 3944) '<unknown>'
    [4000, 4016) 'tmpRef'
    [4064, 4080) 'positiveTrackRef'
    [4128, 4144) 'negativeTrackRef'
    [4192, 4208) 'cxPt'
    [4256, 4272) '<unknown>'
    [4320, 4336) '<unknown>'
    [4384, 4400) '<unknown>'
    [4448, 4464) '<unknown>'
    [4512, 4528) '<unknown>'
    [4576, 4592) '<unknown>'
    [4640, 4656) '<unknown>'
    [4704, 4720) 'theKalmanFitter'
    [4768, 4784) '<unknown>'
    [4832, 4848) 'vtxPos'
    [4896, 4912) '<unknown>'
    [4960, 4976) '<unknown>'
    [5024, 5040) 'positiveP'
    [5088, 5104) 'negativeP'
    [5152, 5168) 'totalP'
    [5216, 5240) 'referencePos'
    [5280, 5304) 'theTrackRefs'
    [5344, 5368) 'theTransTracks'
    [5408, 5432) 'transTracks'
    [5472, 5496) 'distVecXY'
    [5536, 5560) 'distVecXYZ'
    [5600, 5624) 'posTkHitPos'
    [5664, 5688) 'negTkHitPos'
    [5728, 5752) 'theRefTracks'
    [5792, 5816) 'vtx'
    [5856, 5888) 'theTrackHandle'
    [5920, 5952) 'theBeamSpotHandle'
    [5984, 6016) 'vertices'
    [6048, 6080) 'theMagneticFieldHandle'
    [6112, 6144) '<unknown>'
    [6176, 6208) 'kShortP4'
    [6240, 6272) 'lambdaP4'
    [6304, 6336) 'lambdaBarP4'
    [6368, 6400) '<unknown>'
    [6432, 6464) '<unknown>'
    [6496, 6528) '<unknown>'
    [6560, 6592) '<unknown>'
    [6624, 6664) '<unknown>'
    [6720, 6768) 'totalCov'
    [6816, 6864) '<unknown>'
    [6912, 6960) '<unknown>'
    [7008, 7056) '<unknown>'
    [7104, 7152) '<unknown>'
    [7200, 7248) 'vtxCov'
    [7296, 7360) '<unknown>'
    [7392, 7496) 'theAdaptiveFitter'
    [7552, 7672) '<unknown>'
    [7712, 7832) '<unknown>'
    [7872, 8000) 'thePiPlusCand'
    [8032, 8160) 'thePiMinusCand'
    [8192, 8320) 'theProtonCand'
    [8352, 8480) 'theAntiProtonCand'
    [8512, 8672) 'referenceVtx'
    [8704, 8864) 'theVtx'
    [8896, 9072) 'theRecoVertex'
    [9120, 9296) '<unknown>'
    [9344, 9520) '<unknown>'
    [9568, 9792) 'cApp'
    [9824, 10384) '<unknown>'
    [10432, 10992) '<unknown>'
    [11040, 11600) '<unknown>' <== Memory access at offset 11120 is inside this variable
    [11648, 12208) '<unknown>'
    [12256, 12816) '<unknown>'
    [12864, 13424) '<unknown>'
HINT: this may be a false positive if your program uses some custom stack unwind mechanism or swapcontext
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-use-after-scope /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_0_ASAN_X_2017-10-30-2300/src/TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h:90 in GlobalTrajectoryParameters::charge() const
Shadow bytes around the buggy address:
  0x10006b4690e0: f2 f2 f2 f2 f2 f2 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b4690f0: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469100: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469110: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469120: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f2 f2 f2 f2
=>0x10006b469130: f2 f2 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8[f8]f8 f8 f8
  0x10006b469140: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469150: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469160: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
  0x10006b469170: f8 f8 f8 f8 f8 f8 f8 f8 f2 f2 f2 f2 f2 f2 f8 f8
  0x10006b469180: f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8 f8
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==24864==ABORTING
```

**Comment by smuzaffar** [comment]
assign reco

**Comment by perrotta** [comment]
+1
- Addressed by #21195
